### PR TITLE
Issue #13026: Modify the instructions for checks related to record in WriteTagCheck

### PR DESCRIPTION
**Pull request by Sloent**
Issue: #13026 
Modify violation messages associated with 'record'.
Update doc:
image

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheck.java [hunk 1]
```diff
@@ -163,6 +163,10 @@
  * }
  * </pre>
  * <p>
+ * For all the above examples, the handling of 'record' introduced in Java 14 and 'class'
+ * is the same.
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/WriteTagCheck.java:167]: this sounds strange ... as it is also applicable to interfaces and all other target tokens.

**Review by romani — CHANGES_REQUESTED** [review]
Root problem is not clear goal of this ancient Check.
and not clear documentation.

Confusion is that this Check demands:
- if no target javadoc tag give violation
- if tag is present, put violation on tag that match regexp.

I think we need to update description of this Check to describe this.


from doc:
>Requires user defined Javadoc tag to be present in Javadoc comment with defined format.

this is bad wording, as it is not exactly what happening.
proposal: `Requires user defined Javadoc tag to be present in Javadoc comment, tag that match format will be marked as violation.`

>  Property tagSeverity is used for severity of events when the tag exists. 

` Property tagSeverity is used for severity of violations when the tag exists. `


-----

And we need extend example to show it.

**Comment by Sloent** [comment]
> - if no target javadoc tag give violation
> - if tag is present, put violation on tag that match regexp.

So I need to update the documentation for this check and add examples to describe the two requirements above?  Then do I need to keep the other modifications I have made so far?

**Comment by Sloent** [comment]
I got it

**Comment by romani** [comment]
Do not hesitate to close this PR and open new with changes I pointed at https://github.com/checkstyle/checkstyle/pull/13034#pullrequestreview-1404044934

Or simply change things PR.

**Comment by Sloent** [comment]
Maybe I stayed up too hard this month and had a fever of 38.5 degrees last night.  In addition, I have to hand in some course work and a paper for the math school competition when I return from the holiday.  Maybe I can't deal with this issue these days.  After I solve other problems, I will open a new PR according to your requirement.

**Comment by rnveach** [comment]
CI needs to be green. Needs to be rebased.

**Comment by romani** [comment]
I lost reason of this update.
if we do extension of doc we kind should do same for all others modules, but take it for granted that user knows that  interface/class/enum/record are kind the same , they are types that can have javadoc.

**Pull request closed without merging**